src/ModelContextProtocol/Shared/McpSession.cs

```diff
@@ -16,6 +16,12 @@ namespace ModelContextProtocol.Shared;
 /// </summary>
 internal sealed class McpSession : IDisposable
 {
+    /// <summary>
+    /// In-flight request handling, indexed by request ID. The value provides a <see cref="CancellationTokenSource"/>
+    /// that can be used to request cancellation of the in-flight handler.
+    /// </summary>
+    private static readonly ConcurrentDictionary<RequestId, CancellationTokenSource> s_handlingRequests = new();
```

Comment:
Could be wrong, but this feels like it should have been scoped to the current session instance.